src/main/java/com/epam/aidial/core/limiter/RateLimiter.java

```diff
@@ -96,71 +94,125 @@ public Future<LimitStats> getLimitStats(String deploymentName, ProxyContext cont
         if (rateLimit == null) {
-            return limitStats;
+            return;
+        }
+        rateLimit.update(timestamp, limitStats);
+    }
+
+    private void collectRequestLimitStats(ProxyContext context, LimitStats limitStats, long timestamp, String deploymentName) {
+        String requestsPath = getPathToRequests(deploymentName);
+        ResourceDescription resourceDescription = getResourceDescription(context, requestsPath);
+        String json = resourceService.getResource(resourceDescription, true);
+        RequestRateLimit rateLimit = ProxyUtil.convertToObject(json, RequestRateLimit.class);
+        if (rateLimit == null) {
+            return;
         }
-        long timestamp = System.currentTimeMillis();
         rateLimit.update(timestamp, limitStats);
-        return limitStats;
     }
 
     private LimitStats create(Limit limit) {
         LimitStats limitStats = new LimitStats();
-        TokenLimitStats dayTokenStats = new TokenLimitStats();
+
+        ItemLimitStats dayTokenStats = new ItemLimitStats();
         dayTokenStats.setTotal(limit.getDay());
         limitStats.setDayTokenStats(dayTokenStats);
-        TokenLimitStats minuteTokenStats = new TokenLimitStats();
+
+        ItemLimitStats minuteTokenStats = new ItemLimitStats();
         minuteTokenStats.setTotal(limit.getMinute());
         limitStats.setMinuteTokenStats(minuteTokenStats);
+
+        ItemLimitStats hourRequestStats = new ItemLimitStats();
+        hourRequestStats.setTotal(limit.getRequestHour());
+        limitStats.setHourRequestStats(hourRequestStats);
+
+        ItemLimitStats dayRequestStats = new ItemLimitStats();
+        dayRequestStats.setTotal(limit.getRequestDay());
+        limitStats.setDayRequestStats(dayRequestStats);
+
         return limitStats;
     }
 
-    private ResourceDescription getResourceDescription(String deploymentName, ProxyContext context) {
-        String path = getPath(deploymentName);
+    private ResourceDescription getResourceDescription(ProxyContext context, String path) {
         String bucketLocation = BlobStorageUtil.buildUserBucket(context);
         return ResourceDescription.fromEncoded(ResourceType.LIMIT, bucketLocation, bucketLocation, path);
     }
 
-    private RateLimitResult checkLimit(ResourceDescription resourceDescription, Limit limit) {
+    private RateLimitResult checkLimit(ProxyContext context, Limit limit) {
+        long timestamp = System.currentTimeMillis();
+        RateLimitResult tokenResult = checkTokenLimit(context, limit, timestamp);
+        if (tokenResult.status() != HttpStatus.OK) {
+            return tokenResult;
+        }
+        return checkRequestLimit(context, limit, timestamp);
+    }
+
+    private RateLimitResult checkTokenLimit(ProxyContext context, Limit limit, long timestamp) {
+        String tokensPath = getPathToTokens(context.getDeployment().getName());
+        ResourceDescription resourceDescription = getResourceDescription(context, tokensPath);
```

Comment:
you update this state later but never save it back to the storage or I missing something?

Reply:
The state will be updated later by calling the method increase.